pkg/clusterresource/builder.go

```diff
@@ -0,0 +1,492 @@
+			NetworkType:    "OpenShiftSDN",
+			ServiceNetwork: []ipnet.IPNet{*ipnet.MustParseCIDR("172.30.0.0/16")},
+			ClusterNetwork: []installertypes.ClusterNetworkEntry{
+				{
+					CIDR:       *ipnet.MustParseCIDR("10.128.0.0/14"),
+					HostPrefix: 23,
+				},
+			},
+			MachineNetwork: []installertypes.MachineNetworkEntry{
+				{
+					CIDR: *ipnet.MustParseCIDR("10.0.0.0/16"),
+				},
+			},
+		},
+		ControlPlane: &installertypes.MachinePool{
+			Name:     "master",
+			Replicas: pointer.Int64Ptr(3),
+		},
+		Compute: []installertypes.MachinePool{
+			{
+				Name:     "worker",
+				Replicas: &o.WorkerNodesCount,
+			},
+		},
+	}
+
+	o.CloudBuilder.addInstallConfigPlatform(o, installConfig)
+
+	d, err := yaml.Marshal(installConfig)
+	if err != nil {
+		return nil, err
+	}
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      o.getInstallConfigSecretName(),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		Data: map[string][]byte{
+			"install-config.yaml": d,
+		},
+	}, nil
+}
+
+func (o *Builder) generateMachinePool() *hivev1.MachinePool {
+	mp := &hivev1.MachinePool{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "MachinePool",
+			APIVersion: hivev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("%s-worker", o.Name),
+			Namespace: o.Namespace,
+		},
+		Spec: hivev1.MachinePoolSpec{
+			ClusterDeploymentRef: corev1.LocalObjectReference{
+				Name: o.Name,
+			},
+			Name:     "worker",
+			Replicas: pointer.Int64Ptr(o.WorkerNodesCount),
+		},
+	}
+	o.CloudBuilder.addMachinePoolPlatform(o, mp)
+	return mp
+}
+
+func (o *Builder) getInstallConfigSecretName() string {
+	return fmt.Sprintf("%s-install-config", o.Name)
+}
+
+// generatePullSecretSecret returns a Kubernetes Secret containing the pull secret to be
+// used for pulling images.
+func (o *Builder) generatePullSecretSecret() *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      o.getPullSecretSecretName(),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeDockerConfigJson,
+		StringData: map[string]string{
+			corev1.DockerConfigJsonKey: o.PullSecret,
+		},
+	}
+}
+
+// generateSSHPrivateKeySecret returns a Kubernetes Secret containing the SSH private
+// key to be used.
+func (o *Builder) generateSSHPrivateKeySecret() *corev1.Secret {
+	return &corev1.Secret{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Secret",
+			APIVersion: corev1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      o.getSSHPrivateKeySecretName(),
+			Namespace: o.Namespace,
+		},
+		Type: corev1.SecretTypeOpaque,
+		StringData: map[string]string{
+			"ssh-privatekey": o.SSHPrivateKey,
```

Comment:
Should this be a constant maybe? 

We're using it in multiple places in the code base (like in `pkg/install/generate.go`). Maybe this should be added to our hive constants file and referenced by other places?